File: a3/appfrontend/pages/answer/answer.ts
import {Content} from "../content";
import {Question} from "../question/question";

export class Answer {


  private answerQuestion:Question;
  private answerContent:Content;


  constructor( answerQuestion: Question, answerContent: Content) {

    this.answerQuestion = answerQuestion;
    this.answerContent = answerContent;
  }




  get _answerQuestion(): Question {
    return this.answerQuestion;
  }

  set _answerQuestion(value: Question) {
    this.answerQuestion = value;
  }

  get _answerContent(): Content {
    return this.answerContent;
  }

  set _answerContent(value: Content) {
    this.answerContent = value;
  }
}

File: a3/srcbackend/main/java/com/utcn/demo/entity/Answer.java
package com.utcn.demo.entity;


import jakarta.persistence.*;

@Entity
@Table(name = "answers")
public class Answer {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name="id")
    private Long answerId;

    @ManyToOne
    @JoinColumn(name="q_id")
    private Question answerQuestion;

    @OneToOne
    @JoinColumn(name="c_id")
    private Content answerContent;

    public Answer(){

    }

    public Answer( Question answerQuestion, Content answerContent) {
        this.answerId = answerId;
        this.answerQuestion = answerQuestion;
        this.answerContent = answerContent;
    }

    public Long getAnswerId() {
        return answerId;
    }

    public void setAnswerId(Long answerId) {
        this.answerId = answerId;
    }

    public Question getAnswerQuestion() {
        return answerQuestion;
    }

    public void setAnswerQuestion(Question answerQuestion) {
        this.answerQuestion = answerQuestion;
    }

    public Content getAnswerContent() {
        return answerContent;
    }

    public void setAnswerContent(Content answerContent) {
        this.answerContent = answerContent;
    }
}
